### ISSUE TITLE
3.2 release

### ISSUE THREAD

**Issue by xzyfer** [sass/libsass#812]
We've pretty much finished up with the milestone issues for 3.2 (and 3.3!!!). Given that node-sass 2.0 hasnt been released, and the corresponding 3.1 bugs haven't been lodged, I don't think we should jump the gun on a 3.2 release just yet.

How does everyone feel about scheduling 3.2 for some fixed point in the future and just working through as much of the todos as possible until the deadline?

How does 4-6 weeks from the node-sass 2.0 launch sounds?

I'm choosing to tie the release to node-sass' 2.0 since that when 3.1 will hit the majority of our users.

/cc @hcatlin @mgreter @am11

**Comment by mgreter** [comment]
OK for me! I probably will not have much time the next weeks to do some substential work. But I guess there is already enough in the pipeline :smile: I personaly think we should enforce the testing issues a bit more, since it could bite our ass, since we have a few features that are currently not tested.

**Comment by hcatlin** [comment]
Any chance to get #548 in on this release? I've had a couple people contact me via email about it.

**Comment by Jakobud** [comment]
Good news! I dream of a world where Libsass stays up to date with Ruby Sass.

**Comment by michaek** [comment]
@mgreter I've been out of the loop a bit for a few months because of other demands. If you let me know what coverage would be helpful, I'll work on getting that into sass-spec (if that's the test coverage you're referring to).

**Comment by xzyfer** [comment]
@michaek it's be great to get some spec coverage over #548 and generally where `&` can be used as a value.

**Comment by sp90** [comment]
When is 3.2 expected to release - do we have any timespand at this time?

**Comment by xzyfer** [comment]
@sp90 it's hard to say at the moment. You can track the 3.2 progress here: https://github.com/sass/libsass/milestones/3.2

**Comment by mgreter** [comment]
IMO we are pretty much ready once we agree how to merge #927.
So IMHO we probably should plan to release 3.2 in the next days/weeks!?
The only issue left in 3.2 Milestone is https://github.com/sass/libsass/issues/871 and I'm not sure if I'm able to fix this (@xyfer maybe you want to give it a try, documented my findings in that issue). Otherwise I agree that we could re-scope that issue to 3.3 if needed, since it is a minor improvement anyway. @xzyfer are you willing to compile the changelog once again for the next release? Also should we do a beta release first? @am11 what would you're prefered scenario be for your upcoming major node-sass release? Would really like to get the improvements out ASAP! I have already a prepared branch for perl-libsass to support the latest changes. I also already updated some wiki pages to include some of the recent changes!
Just wanted to get this rolling!

**Comment by xzyfer** [comment]
> IMO we are pretty much ready once we agree how to merge #927.

:ship::rocket:

>  So IMHO we probably should plan to release 3.2 in the next days/weeks!?

Sounds great. @hcatlin and @am11 will be happy about this.

> The only issue left in 3.2 Milestone is #871 

I've had a look over the weekend. I know how to fix it but it requires some non-trivial refactoring of the extend visitor. I simply don't have time until April, and it's not a major issue. I'm ok with bumping it to 3.3.

> @xzyfer are you willing to compile the changelog once again for the next release

Yep, I'll knock it out in the next day or so.

> Also should we do a beta release first? 

IIRC node-sass will be doing betas since their releasing a new major version. We should piggy back on this. /cc @am11 

> Just wanted to get this rolling!
 :rocket: :boom:

**Comment by am11** [comment]
Its an excellent news! :sunglasses: 
@xzyfer, I am thinking about a preview release (alpha) with libsass v3.1.0 to relax the io.js situation, once we get v3.2.0 ready, we can add more features and land on beta. Who knows we might be able to get https://github.com/sass/node-sass/pull/644 ready by @matryo by that time. 8-)

**Comment by mgreter** [comment]
:golf: I guess we're ready, since I merged #927 :exclamation:
Some code can be considered "hotfixes", but better than failing specs IMO!
So I pass the :soccer: to you to compile the changelog for the release :smiley:

**Comment by am11** [comment]
:checkered_flag: :tada:

**Comment by mgreter** [comment]
BTW. we're down to 21 open issues and closed 83 issues with [Milestone 3.2] [1] :100: 

[1]: https://github.com/sass/libsass/milestones/3.2

**Comment by am11** [comment]
Yes and only 9 bugs to terminate: https://github.com/sass/libsass/issues?q=is%3Aopen+is%3Aissue+label%3Abug. :smiley: :+1:

**Comment by xzyfer** [comment]
Also worth a mention, we added 66 new specs with ~270 extra assertions and skip 26 less specs.

```
master
799 runs, 2109 assertions, 0 failures, 0 errors, 96 skips

3.1.0
733 runs, 1833 assertions, 0 failures, 0 errors, 122 skips
```

**Comment by hcatlin** [comment]
How about a Camp Sass release in like a month? Do preview till then?

**Comment by mgreter** [comment]
@xzyfer @hcatlin I took the liberty to create a (non public) release draft!

**Comment by xzyfer** [comment]
Thanks @mgreter! Sorry I haven't had the time to do it yet, will try to do so shortly

**Comment by mgreter** [comment]
OK, from my side we are ready to ship anytime now ...

**Comment by xzyfer** [comment]
OK I've updated the release notes.

**Comment by mgreter** [comment]
I've updated the wiki for the C-API.
Now there are basically only two minor issues open:
- https://github.com/sass/libsass/pull/935 - don't like to have version hardcoded in the git repo
- https://github.com/sass/sass-spec/pull/270 - would increase code coverage by quite a bit 

I guess the second one if not really urgent (nice to have), since I know it works correctly on my maschine :wink: But I would like to merge the first one, if nothing strictly speaks against it.

**Comment by mgreter** [comment]
Oh, and I guess we're going first with an alpha release (`3.2.0-alpha`)?

**Comment by xzyfer** [comment]
I've added some formatting to the release, emphasising the cool new features.

I think we can release a `3.2.0-beta`. Alpha doesn't make a lot of sense since very few people will use Libsass until it's hits the `node-sass@3.0.0-beta`

**Comment by xzyfer** [comment]
IMO :ship: those two outstanding issues.

I suspect sass-spec will largely be re-written soon when the sass-spec directory structure changes.

**Comment by mgreter** [comment]
OK, will now clean the commit for the new sass-specs and check that CI still works. There will be about 3000 new spec tests, if you count them individually :smile:. If OK for you, I will ship `3.2.0-alpha.1` right after CI gives green light after the two merges !??

**Comment by xzyfer** [comment]
`3.2.0-beta` ;) but otherwise :+1:

SassC also needs a new release as well.

**Comment by mgreter** [comment]
Beta release `3.2.0-beta.1` is out :rocket:
```
3955 runs, 10701 assertions, 0 failures, 0 errors, 388 skips
```

**Comment by am11** [comment]
:+1: :tada:
